pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4214 @@
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          InstanceName (:class:`~pywbem.CIMInstanceName`):
+            The instance path of the instance to be retrieved.
+            If this object does not specify a namespace, the default namespace
+            of the connection is used.
+            Its `host` attribute will be ignored.
+
+          LocalOnly (:class:`py:bool`):
+            Controls the exclusion of inherited properties from the returned
+            instance, as follows:
+
+            * If `False`, inherited properties are not excluded.
+            * If `True`, inherited properties are basically excluded, but the
+              behavior may be WBEM server specific.
+            * If `None`, the :term:`DSP0200` defined server  default of `True`
+              is used.
+
+          IncludeQualifiers (:class:`py:bool`):
+            Indicates that qualifiers are to be included in the returned
+            instance, as follows:
+
+            * If `False`, qualifiers are not included.
+            * If `True`, qualifiers are included if the WBEM server implements
+              support for this parameter.
+            * If `None`, the :term:`DSP0200` defined server default `False`
+              is used.
+
+            This parameter has been deprecated in :term:`DSP0200`. Clients
+            cannot rely on qualifiers to be returned in this operation.
+
+
+          IncludeClassOrigin (:class:`py:bool`):
+            Indicates that class origin information is to be included on each
+            property in the returned instance, as follows:
+
+            * If `False`, class origin information is not included.
+            * If `True`, class origin information is included.
+            * If `None`, uses :term:`DSP0200` server default `False`.
+
+
+          PropertyList (:term:`string` or :term:`py:iterable` of :term:`string`):
+            An iterable specifying the names of the properties (or a string
+            that defines a single property) to be included in the returned
+            instance (case independent).
+
+            An empty iterable indicates to include no properties.
+
+            If `None`, all properties are included.
+
+        Returns:
+
+            A :class:`~pywbem.CIMInstance` object that is a representation of
+            the retrieved instance.
+            Its `path` attribute is a :class:`~pywbem.CIMInstanceName` object
+            with its attributes set as follows:
+
+            * `classname`: Name of the creation class of the instance.
+            * `keybindings`: Keybindings of the instance.
+            * `namespace`: Name of the CIM namespace containing the instance.
+            * `host`: `None`, indicating the WBEM server is unspecified.
+            TODO, host. Confirm what our mock server does.
```

Comment:
The CIM objects returned from the operation methods of WBEMConnection always have namespace set and host unset, regardless of how the underlying CIM-XML operations define their result objects. So this is the contract for the operation methods of FakedWBEMConnection. For the providers, we could theoretically establish a different contract, but I suggest that we establish this as a rule also for the providers.

Reply:
Confrimed that we clear the host in add_objects, and do not set it in CreateInstance so that it is not set in the repository.  Also, in _get_instance.  Added test for confirmation and modified doc slightly. Removed TODO

DONE